storage/src/mmr/mem.rs

```diff
@@ -26,22 +32,28 @@ impl<H: CHasher> Mmr<H> {
         Self {
             hasher: H::new(),
             nodes: Vec::new(),
+            oldest_remembered_pos: 0,
         }
     }
 
     pub fn size(&self) -> usize {
-        self.nodes.len()
+        self.nodes.len() + self.oldest_remembered_pos as usize
     }
 
     /// Return a new iterator over the peaks of the MMR.
     fn peak_iterator(&self) -> PeakIterator {
-        PeakIterator::new(self.nodes.len() as u64)
+        PeakIterator::new(self.size() as u64)
+    }
+
+    // Return the position of the element given its index in the current nodes vector.
+    fn index_to_pos(&self, index: usize) -> u64 {
```

Comment:
It seems odd that we use `u64` for position but the vector may only hold `u32::MAX` elements?

As convenient as it might be to restrict to `64-bit architecture`, WASM is only 32-bit rn so we might need to think about this a bit.

Reply:
I wanted position to be implementation independent since they are incorporated into hashing.  If it varies by architecture or implementation then proving becomes architecture and implementation specific.

Reply:
I agree that position should be implementation independent.

I think we should consider making `index` a `u64` as well (as position, to me, implies index must also be at least u64)?

Reply:
This might just matter for the memory version of this (as the stateful version already uses u64 indicies)

Reply:
yeah index is an implementation specific concept, so thought usize made the most sense here, but don't feel strongly.

Reply:
I think we should change it to `u64` and leave it to the implementation to handle that.

Reply:
Once that is done, we are probably pretty close to merging.

Reply:
done.  working on the serialize/deserialize but happy to have that go in a later PR.